### PR TITLE
Upgrade frameworks version of golangci-lint

### PR DESCRIPTION
**Pull request by willbeason**
Use the same linters we use in gatekeeper/ for consistency.

Remove usage of deprecated/now-unsupported linters.

Make changes to make linting pass.

Signed-off-by: Will Beason <willbeason@google.com>

### CODE REVIEW

File: constraint/pkg/client/backend.go [hunk 1]
```diff
@@ -65,7 +66,7 @@ func (b *Backend) NewClient(opts ...Opt) (*Client, error) {
 	}
 	for _, field := range c.allowedDataFields {
 		if !validDataFields[field] {
-			return nil, errors.Errorf("Invalid data field %s", field)
+			return nil, fmt.Errorf("Invalid data field %s", field)
```
Comment (maxsmythe) [inline constraint/pkg/client/backend.go:69]: nit: lowercase first letter?

File: constraint/pkg/client/drivers/local/local.go [hunk 2]
```diff
@@ -107,20 +109,20 @@ func copyModules(modules map[string]*ast.Module, filter string) map[string]*ast.
 
 func (d *driver) checkModuleName(name string) error {
 	if name == "" {
-		return errors.Errorf("Module name cannot be empty")
+		return fmt.Errorf("Module name cannot be empty")
```
Comment (maxsmythe) [inline constraint/pkg/client/drivers/local/local.go:112]: nit here and below... lowercase first letter?

**Review by maxsmythe — APPROVED** [review]
LGTM with nits

**Review by sozercan — APPROVED** [review]
Thank you! LGTM pending @maxsmythe's comments

File: constraint/pkg/client/backend.go (hunk 1 above)
Comment (willbeason) [reply constraint/pkg/client/backend.go:69]: Done.

File: constraint/pkg/client/drivers/local/local.go (hunk 2 above)
Comment (willbeason) [reply constraint/pkg/client/drivers/local/local.go:112]: Done. Fixed about a hundred such similar cases.

File: constraint/pkg/regorewriter/sources.go [hunk 3]
```diff
@@ -91,18 +91,19 @@ func (s *Sources) Reparent(old, new string) error {
 
 // Write will write the sources to the filesystem.
 func (s *Sources) Write() error {
-	return s.forAll(func(module sourceFile) error {
+	// TODO: Determine if _ is intended to be used in anything.
```
Comment (sozercan) [inline constraint/pkg/regorewriter/sources.go:94]: open TODO here

Comment (willbeason) [reply constraint/pkg/regorewriter/sources.go:94]: I talked with Max about this - we weren't sure a great way to mark that this is obviously a bug in the code. It isn't worth the time to fix it and verify the original intent as it was written two years ago, is untested, and isn't used in any production code.

The intent is to leave a note for someone in the future (who presumably will care about the functionality of that code) that there is obviously a problem here, but we don't really know for sure what will fix it.

**Comment by sozercan** [comment]
@willbeason good to merge?

**Pull request merged**